build.sbt

```diff
@@ -57,21 +58,15 @@ lazy val client = (project in file("clients/java"))
   .settings(
     name := s"$artifactNamePrefix-client",
     commonSettings,
-    libraryDependencies ++= Seq(
-      "com.fasterxml.jackson.core" % "jackson-annotations" % jacksonVersion,
-      "com.fasterxml.jackson.core" % "jackson-core" % jacksonVersion,
-      "com.fasterxml.jackson.core" % "jackson-databind" % jacksonVersion,
-      "com.fasterxml.jackson.datatype" % "jackson-datatype-jsr310" % jacksonVersion,
-      "org.openapitools" % "jackson-databind-nullable" % openApiToolsJacksonBindNullableVersion,
-      "com.google.code.findbugs" % "jsr305" % "3.0.2",
-      "jakarta.annotation" % "jakarta.annotation-api" % "1.3.5" % Provided,
-
-      // Test dependencies
-      "junit" %  "junit" % "4.13.2" % Test,
-      "org.junit.jupiter" % "junit-jupiter" % "5.9.2" % Test,
-      "net.aichler" % "jupiter-interface" % JupiterKeys.jupiterVersion.value % Test,
-    ),
-
+    javaCheckstyleSettings(file("dev") / "checkstyle-config.xml"),
+    libraryDependencies ++=
+      Dependencies.findBugs ++
+      Dependencies.guava ++ 
+      Dependencies.hadoop ++ 
+      Dependencies.jackson(includeAnnotations = true) ++
+      Dependencies.Jakarta.annotation ++ 
```

Comment:
While I like this clustering of libraries and simplifying the high level view of dependencies, I don't like the loss of information of which dependencies are for prod vs test. Can you somehow indicate that in this high level glance?

Reply:
I think what we can do for junit4 for for example is to write something like that:

```scala
     Dependencies.Jakarta.annotation ++ 
      Dependencies.junit4.map(_ % Test),
```

if we want to ensure we don't define the scope of dependencies in `Dependencies.scala` but only in `build.sbt`.

An alternative approach could be to create an object inside Dependencies and have something like so:

```scala
     Dependencies.Compile.Jakarta.annotation ++ 
      Dependencies.Test.junit4 ++
      Dependencies.Provided.javax
```

I like this last idea, using the nested support of Scala objects as a form of namespace to be more explicit about the scope of a dependency?

Reply:
While i like the structure of `Dependencies.Provided.javax`, managing it is more work. If something needs to be converting from compile to provided scope, then 2 locations need to change - Dependencies AND build.sbt. 

In general, there should be a separate of concerns
- Dependencies.scala - defines the all the dependent libraries that any module in SBT can depend on. It will ensure that all modules depend on the correct and consistent version of each library. It should not care about which SBT module is using which dependency in what way (compile or test).
- Versions.scala - defines the versions cleanly. should not care about which SBT module is using that version in what way.
- build/sbt - combines the information above to define a module. So build.sbt has the responsibilty to define whether a particular dependency is compile, or test, or provided.

So please update this PR with this approach `Dependencies.junit4.map(_ % Test),`

Reply:
@edmondop any thoughts?

Reply:
That makes sense. I'll fix this today

Reply:
Fixed with [`49bf991` (#37)](https://github.com/unitycatalog/unitycatalog/pull/37/commits/49bf991ceff2a7e89e7cbf05e5829c9b2c46c458)